WordPress/src/main/java/org/wordpress/android/userflags/provider/UserFlagsProvider.kt

```diff
@@ -0,0 +1,119 @@
+package org.wordpress.android.userflags.provider
+
+import android.database.Cursor
+import android.net.Uri
+import org.wordpress.android.WordPress
+import org.wordpress.android.fluxc.store.SiteStore
+import org.wordpress.android.provider.query.QueryContentProvider
+import org.wordpress.android.provider.query.QueryResult
+import org.wordpress.android.ui.prefs.AppPrefs.DeletablePrefKey
+import org.wordpress.android.ui.prefs.AppPrefs.UndeletablePrefKey
+import org.wordpress.android.ui.prefs.AppPrefsWrapper
+import org.wordpress.android.util.publicdata.JetpackPublicData
+import org.wordpress.android.util.signature.SignatureNotFoundException
+import org.wordpress.android.util.signature.SignatureUtils
+import javax.inject.Inject
+
+class UserFlagsProvider : QueryContentProvider() {
+    @Inject lateinit var appPrefsWrapper: AppPrefsWrapper
+    @Inject lateinit var siteStore: SiteStore
+    @Inject lateinit var signatureUtils: SignatureUtils
+    @Inject lateinit var queryResult: QueryResult
+    @Inject lateinit var jetpackPublicData: JetpackPublicData
+
+    private val userFlagsKeysSet: Set<String> = setOf(
+            DeletablePrefKey.READER_TAG_NAME.name,
```

Comment:
Good catch! With our initial approach (handling each flag manually) I had removed the other `READER_TAG_` flags because they were all assembled using the value from `READER_TAG_NAME`:

```
    public static void setReaderTag(ReaderTag tag) {
        if (tag != null && !TextUtils.isEmpty(tag.getTagSlug())) {
            setString(DeletablePrefKey.READER_TAG_NAME, tag.getTagSlug());
            setInt(DeletablePrefKey.READER_TAG_TYPE, tag.tagType.toInt());
```

But with our new approach of getting the raw values from Prefs, they would be lost. I'll fix it 👍